app/src/components/market/common/xdai_bridge_modal/index.tsx

```diff
@@ -0,0 +1,222 @@
+    width: calc(100%);
+    right: 1px;
+  }
+`
+
+const ChainText = styled.div`
+  text-align: start;
+`
+const BalanceText = styled.div<{ disabled: boolean }>`
+  text-align: end;
+  width: fit-content;
+  color: ${({ disabled, theme }) => (disabled ? theme.colors.textColorLighter : theme.colors.clickable)};
+  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
+  &:hover {
+    ${({ disabled, theme }) => !disabled && `color:${theme.colors.primaryLight}`};
+  }
+`
+
+const MainnetWrapper = styled.div`
+  margin-bottom: 12px;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+`
+const XDaiWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+`
+const BalanceWrapper = styled.div<{ isOpen: boolean }>`
+  flex-wrap: wrap;
+  ${props => (!props.isOpen ? 'display:none' : 'display: flow-root')};
+`
+const Warning = styled.div`
+  color: ${({ theme }) => theme.colors.red};
+  opacity: 80%;
+  margin-left: 8px;
+`
+const WarningWrapper = styled.div`
+  display: flex;
+  margin-top: 8px;
+`
+
+const TextFieldCustomPlace = styled(TextfieldCustomPlaceholder)<{ error: boolean }>`
+  margin-top: 20px;
+  ${props => (props.error ? `border:1px solid ${props.theme.colors.red} !important` : '')};
+  span {
+    margin-right: 0px;
+  }
+`
+
+const TransferButton = styled(ButtonRound)`
+  margin-top: 20px;
+  width: 100%;
+`
+
+const PoweredByStakeWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  margin-top: 20px;
+  justify-content: center;
+`
+const StakeText = styled.a`
+  margin-left: 8px;
+  font-size: 12px;
+  font-weight: 500;
+  color: ${({ theme }) => theme.colors.clickable};
+  line-height: 12px;
+  &:hover {
+    ${({ theme }) => `color:${theme.colors.primaryLight}`};
+  }
+`
+
+export const XdaiBridgeTransfer = (props: Prop) => {
+  const [amountToDisplay, setAmountToDisplay] = useState<string>('')
+  const [amount, setAmount] = useState<BigNumber>(new BigNumber(0))
+
+  const { networkId } = useConnectedWeb3Context()
+
+  const [transferState, setTransferState] = useState<boolean>(false)
+  const errorEval = amount.gt(Zero) && Number(formatBigNumber(amount, 18, 2)) < 10 && networkId != 1
+
+  const {
+    daiBalance,
+    fetchBalance,
+    numberOfConfirmations,
+    transactionHash,
+    transactionStep,
+    transferFunction,
+    xDaiBalance,
+  } = useXdaiBridge(amount)
+  return (
+    <>
+      <BridgeWrapper isOpen={props.open}>
+        <BalanceWrapper isOpen={!transferState}>
+          <MainnetWrapper>
+            <ChainText>Mainnet</ChainText>
+            <BalanceText
+              disabled={networkId != 1}
+              onClick={() => {
+                if (daiBalance.eq(Zero) || networkId != 1) return
+                setAmount(daiBalance)
+                setAmountToDisplay(formatBigNumber(daiBalance, 18))
+              }}
+            >
+              {formatBigNumber(daiBalance, 18)} DAI
```

Comment:
encapsulate the Dai token into a token object and the use daiToken.decimals here